neo4j-backup/templates/neo4j-cronjob.yaml

```diff
@@ -0,0 +1,98 @@
+{{- template "neo4j.backup.checkIfSecretExistsOrNot" . -}}
+apiVersion: batch/v1
+kind: CronJob
+metadata:
+  name: "{{ include "neo4j.fullname" . }}"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ include "neo4j.fullname" . | quote }}
+    {{- include "neo4j.labels" $.Values.neo4j.labels | indent 4 }}
+spec:
+  schedule: {{ $.Values.neo4j.jobSchedule | default "* * * * *" | quote }}
+  concurrencyPolicy: Forbid
+  successfulJobsHistoryLimit: {{ $.Values.neo4j.successfulJobsHistoryLimit | default 3 }}
+  failedJobsHistoryLimit: {{ $.Values.neo4j.failedJobsHistoryLimit | default 1 }}
+  jobTemplate:
+    spec:
+      backoffLimit: {{ $.Values.neo4j.backoffLimit | default 6 }}
+      template:
+        metadata:
+          annotations:
+            {{- include "neo4j.annotations" $.Values.neo4j.podAnnotations | indent 12 }}
+          labels:
+            {{- include "neo4j.labels" $.Values.neo4j.podLabels | indent 12 }}
+        spec:
+          {{- if .Values.serviceAccountName }}
+          serviceAccountName: {{ .Values.serviceAccountName }}
+          {{- /* explicitly mount token because some service accounts disable automount-by-default and require explicit opt-in */}}
+          automountServiceAccountToken: true
+          {{- end }}
+          restartPolicy: Never
+          securityContext: {{ .Values.securityContext | toYaml  | nindent 12 }}
+          containers:
+            - name: graph-backup
+              image: {{ .Values.neo4j.image }}:{{ .Values.neo4j.imageTag }}
+              imagePullPolicy: Always
+              resources:
+                requests:
+                  ephemeral-storage: {{ .Values.resources.requests.ephemeralStorage | default "4Gi" }}
+                limits:
+                  ephemeral-storage: {{ .Values.resources.limits.ephemeralStorage | default "5Gi" }}
+              env:
+                - name: ADDRESS
+                  value: {{ required "You must specify a ADDRESS variable in values.yaml in the format <host>:<port> . Set it via --set backup.address" .Values.backup.address  | trim }}
+                - name: DATABASE
+                  value: {{ required "You must specify a DATABASE variable in values.yaml. . Set it via --set backup.database" .Values.backup.database | trim }}
```

Comment:
I think this could be optional and default to "*"

Reply:
I think specifying the correct database must be compulsory and not optional...specifying * can lead to unwanted bigger backups.

Reply:
I would argue that it is better to have bigger backups than forget to backup. Someone could add a new DB and forget to update the script

Reply:
okay i will make it optional.